.github/workflows/ci.yml

```diff
@@ -18,6 +18,26 @@ jobs:
   build:
     runs-on: ubuntu-latest
     timeout-minutes: 25
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - goos: 'windows'
+            goarch: 'amd64'
+          - goos: 'windows'
+            goarch: 'arm64'
+
+          - goos: 'darwin'
+            goarch: 'amd64'
+          - goos: 'darwin'
+            goarch: 'arm64'
+
+          - goos: 'linux'
+            goarch: 'amd64'
+          - goos: 'linux'
+            goarch: 'arm64'
+          - goos: 'linux'
+            goarch: 'arm'
```

Comment:
I'm fine with this shorter list for a PR-based workflow, but I think we should continue building all platforms on push to `main` (i.e. after PR is merged). Unfortunately both of these are now handled by the same workflow - I'm not sure if there's any better solution other than duplicating the workflow into a separate YAML? 😞

Reply:
We can use reusable workflows to split out the core parts of this one workflow, then have the scheduled events or triggers handle executing each extracted workflow. Another approach would be splitting up the goreleaser file into multiple files and having different scheduled actions.

There will be duplication any way we go, but this might not really be a limitation per se. We have different purposes for each part, and clearly defined 'rules' on when they should run. Splitting these out into 'unit' or 'acceptance' with scheduled or triggered would reduce resource costs and time the user is waiting.